src/ncp/ncp_base.cpp

```diff
@@ -6683,35 +6680,30 @@ otError NcpBase::InsertPropertyHandler_THREAD_OFF_MESH_ROUTES(uint8_t aHeader, s
                        &flags
                    );
 
-    if (parsedLength > 0)
-    {
-        routeConfig.mPrefix.mPrefix = *addrPtr;
-        routeConfig.mStable = stable;
-        routeConfig.mPreference = FlagByteToExternalRoutePreference(flags);
+    VerifyOrExit(parsedLength > 0, spinelError = SPINEL_STATUS_PARSE_ERROR);
 
-        error = otBorderRouterAddRoute(mInstance, &routeConfig);
+    routeConfig.mPrefix.mPrefix = *addrPtr;
+    routeConfig.mStable = stable;
+    routeConfig.mPreference = FlagByteToExternalRoutePreference(flags);
 
-        if (error == OT_ERROR_NONE)
-        {
-            error = SendPropertyUpdate(
-                        aHeader,
-                        SPINEL_CMD_PROP_VALUE_INSERTED,
-                        aKey,
-                        aValuePtr,
-                        aValueLen
-                    );
-        }
-        else
-        {
-            error = SendLastStatus(aHeader, ThreadErrorToSpinelStatus(error));
-        }
-    }
-    else
+    error = otBorderRouterAddRoute(mInstance, &routeConfig);
+    VerifyOrExit(error == OT_ERROR_NONE, spinelError = ThreadErrorToSpinelStatus(error));
+
+    error = SendPropertyUpdate(
```

Comment:
We would send the `LastStatus` at `exit` only if `spinelError` is set to other than `STATUS_OK`.  `SendPropertyUpdate()`'s returned value is stored in `error` (`spinelError` should remain `STATUS_OK).

We have two error variables: 
- `spinelError` which tracks the error we would want to report back in response to the spinel command. 
- `otError error` which we finally return from the method  and should indicate success/failure of sending/queuing  the spinel response (independent of the response being for success case or a `LastStatus` spinel error report).